lib/internal/modules/esm/loader.js

```diff
@@ -329,7 +329,7 @@ class ModuleLoader {
    * @returns {ModuleJobBase}
    */
   getModuleJobForRequire(specifier, parentURL, importAttributes) {
-    assert(getOptionValue('--experimental-require-module'));
+    assert(getOptionValue('--experimental-require-module') || getOptionValue('--experimental-detect-module'));
```

Comment:
Why is this added? This shouldn’t be hit when requiren module isn’t enabled.

Reply:
removed, that might have been added while I was testing with the flipped default value for the detect-module flag.